### PR TITLE
AIAssistant Suggestions

### PR DESCRIPTION
**Pull request by vladtar**
Added prompts suggesting to a user potential use cases they might be interested in.

## Security

REMINDER: All file contents are public.

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc.
  - [x] Please verify that you double checked that .storybook/preview.js does not contain your participant access key details. 
  - [x] There are no temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] These changes do not introduce any security risks, or any such risks have been properly mitigated.

Describe briefly what security risks you considered, why they don't apply, or how they've been mitigated.

### Testing
- [ ] MyDataHelps iOS app
- [ ] MyDataHelps Android app
- [x] [MyDataHelps website](mydatahelps.org)

### Documentation
- [x] I have added relevant Storybook updates to this PR as well.
- [ ] If this feature requires a developer doc update, tag @CareEvolution/api-docs.

Consider "Squash and merge" as needed to keep the commit history reasonable on `main`.

## Reviewers

Assign to the appropriate reviewer(s). Minimally, a second set of eyes is needed ensure no non-public information is published. Consider also including:
- Subject-matter experts
- Style/editing reviewers
- Others requested by the content owner

image

## Summary by CodeRabbit

- **New Features**
  - Introduced a `showSuggestions` feature in the AIAssistant, allowing users to toggle the display of suggestions.
  - Expanded the available tools in the AIAssistant component.
  - Added a new `NoSuggestions` story to demonstrate the assistant without suggestions.
  - Enhanced the AI assistant with new health-related suggestions in multiple languages.

- **Enhancements**
  - Improved the Chat component to handle and display suggestions interactively.

- **Style Updates**
  - Added new styles for chat suggestions to enhance visual presentation.

### CODE REVIEW

File: src/components/container/AIAssistant/AIAssistant.tsx [hunk 1]
```diff
@@ -36,18 +37,27 @@ export default function (props: AIAssistantProps) {
...
 
     useEffect(() => {
         if (assistantRef.current === undefined) {
             assistantRef.current = new MyDataHelpsAIAssistant(props.baseUrl, props.additionalInstructions, [new CustomEventTrackerCallbackHandler()], props.tools, props.appendTools);
         }
+
+        if (props.showSuggestions) {
+            setSuggestions(defaultSuggestions
+                .sort(() => 0.5 - Math.random())
+                .slice(0, 3)
```
Comment (vladtar) [inline src/components/container/AIAssistant/AIAssistant.tsx:52]: Pick 3 random suggestions.

File: src/components/container/AIAssistant/AIAssistant.tsx [hunk 2]
```diff
@@ -85,6 +95,8 @@ export default function (props: AIAssistantProps) {
                         if (image && !image.startsWith("error")) {
                             addMessage(streamEvent.run_id, `data:image/png;base64,${image}`, "image");
                         }
+
+                        setSuggestions(["Save the graph to my files"]);
```
Comment (vladtar) [inline src/components/container/AIAssistant/AIAssistant.tsx:99]: After the user generates a graph prompt them to save it to their files.

**Review by ClaysGit — APPROVED** [review]
Seems like there's 2 logical steps from here:
1. User-provided suggestions - In line with Vik's feedback on the AI integration inventory doc, can help targeted adopter / use case
2. Localization support (default and user-provided suggestions)

File: src/components/presentational/Chat/Chat.tsx [hunk 3]
```diff
@@ -84,6 +86,13 @@ export default function (props: ChatProps) {
                     </div>
                 </div>
             </div>
+            {props.suggestions && props.onSuggestionSelected && (
+                <div className="mdhui-chat-suggestions">
```
Comment (reneefromhold) [inline src/components/presentational/Chat/Chat.tsx:90]: When the view port is small mobile or large tablet the presentation of the suggestions looks strangle side by side. Changing the flex-direction on .mdhui-chat-suggestions = column, imho, makes it more visually appealing

File: src/components/presentational/Chat/Chat.tsx [hunk 4]
```diff
@@ -21,6 +21,8 @@ export interface ChatProps {
     onSendMessage: (newMessage: string) => void;
     loading?: string;
     inputDisabled?: boolean;
+    suggestions?: string[];
```
Comment (reneefromhold) [inline src/components/presentational/Chat/Chat.tsx:24]: Add a story for suggestions or include being able to update it in the default story

File: src/components/container/AIAssistant/AIAssistant.stories.tsx [hunk 5]
```diff
@@ -28,6 +28,13 @@ export default {
...
+        showSuggestions: {
+            control: 'boolean',
+            description: 'If turned on the assistant will show suggestions to the user.',
+            defaultValue: {
+                summary: true
+            }
+        },
         appendTools: {
             control: false,
             description: 'If turned on the tools passed in the tools prop will be appended to the default tools.',
```
Comment (reneefromhold) [inline src/components/container/AIAssistant/AIAssistant.stories.tsx:40]: Would be helpful to add a blurb on what the default tools are.

**Review by reneefromhold — APPROVED** [review]
This feature is awesome. 
I left some comments and ask if you could
1) Add a jsdoc comment on the component and export it as ...default and named. See here for [comment example ](https://github.com/CareEvolution/MyDataHelpsUI/blob/539663ff03c710e7867d2b5cc1b4d05faa5f959f/src/components/container/BasicPointsForBadges/BasicPointsForBadges.tsx#L27)
2) Display that comment as the default story description. See [here for example](https://github.com/CareEvolution/MyDataHelpsUI/blob/539663ff03c710e7867d2b5cc1b4d05faa5f959f/src/components/container/BasicPointsForBadges/BasicPointsForBadges.stories.tsx#L19).  This description will be exported along with the component in the npm package

File: src/components/presentational/Chat/Chat.tsx (hunk 3 above)
Comment (vladtar) [reply src/components/presentational/Chat/Chat.tsx:90]: Screenshot? Looks reasonable to me. Was also following the chatGPT model they have things lined up the same way.

File: src/components/presentational/Chat/Chat.tsx (hunk 4 above)
Comment (vladtar) [reply src/components/presentational/Chat/Chat.tsx:24]: Added.

File: src/components/container/AIAssistant/AIAssistant.stories.tsx (hunk 5 above)
Comment (vladtar) [reply src/components/container/AIAssistant/AIAssistant.stories.tsx:45]: This parameter doesn't make too much sense in a story context. Other tools would be passed in here programmatically, and this is where you would say if you wanted the new tools to be the only tools available or whether you want the new tools to be added to the default ones. The default tools are just the list that shows up under the tools parameter here.

File: src/components/presentational/Chat/Chat.tsx (hunk 3 above)
Comment (reneefromhold) [reply src/components/presentational/Chat/Chat.tsx:93]: @vladtar Thanks for drawing my attention to chatgpt on mobile.  Chatgpt will display wider text and allows the scrolling left/right through suggestions. The multiple lines of the current format are what stand out as clunky. 
There is also a neat approach to suggestions within ChatGpt where a generalized question is in bold - What is my average - for example - and below in sub text is a more direct phrase like, sleep each night/heart rate this week/steps per day?

Comment (reneefromhold) [reply src/components/presentational/Chat/Chat.tsx:93]: image

**Pull request merged**